packages/amplify-cli/src/extensions/amplify-helpers/update-tracked-files.ts

```diff
@@ -0,0 +1,80 @@
+import * as path from 'path';
+import fs from 'fs-extra';
+import {
+  $TSAny, JSONUtilities, pathManager, stateManager,
+} from 'amplify-cli-core';
+
+const {
+  readJson,
+} = JSONUtilities;
+
+/**
+ * Updates Cognito files that are tracked so that the diff is detected for an `amplify push`
+ */
+export const updateCognitoTrackedFiles = async (): Promise<void> => {
+  const currentCloudBackendDir = pathManager.getCurrentCloudBackendDirPath();
+  const localBackendDir = pathManager.getBackendDirPath();
+  const amplifyMeta = stateManager.getMeta();
+  const cognitoResource = stateManager.getResourceFromMeta(amplifyMeta, 'auth', 'Cognito', undefined, false);
+
+  if (!fs.existsSync(currentCloudBackendDir) || !cognitoResource) {
+    return;
+  }
+
+  const { resourceName } = cognitoResource;
+
+  if (await detectCognitoDiff(currentCloudBackendDir, localBackendDir, resourceName)) {
+    await addExtraLineToCliInputsJson(localBackendDir, resourceName);
+  }
+};
+
+const detectCognitoDiff = async (
+  currentCloudBackendDir: string,
+  localBackendDir: string,
+  resourceName: string,
+): Promise<boolean> => detectCognitoAttributesRequireVerificationBeforeUpdateDiff(
+  currentCloudBackendDir,
+  localBackendDir,
+  resourceName,
+);
+
+const detectCognitoAttributesRequireVerificationBeforeUpdateDiff = async (
```

Comment:
It seems you have two things going on here; one very generic thing "checking if a local update is not present in the cloud" and one very specific thing "checking if a cognito setting is a specific value that differs locally vs in the cloud" . I'm guessing the original issue is more generic "any setting locally should also reflect in the cloud & trigger an update", otherwise it seems like we are patching a bug with that setting specifically. Can you explain why we only check this specific setting; and why we don't provide a more high level diff check

Reply:
resolving this as we only care about this single property; and there are other issues with diff tracking